CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,157 @@
+# How to Contribute to Jaeger
```

Comment:
Originally I appended "Operator" to this title and in a few other places. But decided to revert as the instructions basically relate to all jaeger repos. They are just here for convenience.